pkg/loop/internal/core/services/capability/capabilities.go

```diff
@@ -331,73 +331,78 @@ func (c *callbackExecutableServer) UnregisterFromWorkflow(ctx context.Context, r
 	req, err := pb.CapabilityRequestFromProto(reqpb)
 	if err != nil {
 		return fmt.Errorf("could not unmarshal capability request: %w", err)
 	}
 
-	responseCh, err := c.impl.Execute(server.Context(), req)
+	var responseMessage *capabilitiespb.ResponseMessage
+	response, err := c.impl.Execute(server.Context(), req)
 	if err != nil {
-		return fmt.Errorf("error executing capability request: %w", err)
-	}
-
-	err = server.Send(&capabilitiespb.ResponseMessage{
-		Message: &capabilitiespb.ResponseMessage_Ack{
-			Ack: &emptypb.Empty{},
-		},
-	})
-	if err != nil {
-		return fmt.Errorf("error sending ack: %w", err)
-	}
-
-	for resp := range responseCh {
-		msg := &capabilitiespb.ResponseMessage{
+		responseMessage = &capabilitiespb.ResponseMessage{
 			Message: &capabilitiespb.ResponseMessage_Response{
-				Response: pb.CapabilityResponseToProto(resp),
+				Response: &capabilitiespb.CapabilityResponse{Error: err.Error()},
 			},
 		}
-		if err = server.Send(msg); err != nil {
-			return fmt.Errorf("error sending response for execute request %s: %w", reqpb, err)
+	} else {
+		responseMessage = &capabilitiespb.ResponseMessage{
+			Message: &capabilitiespb.ResponseMessage_Response{
+				Response: pb.CapabilityResponseToProto(response),
+			},
 		}
 	}
 
+	if err = server.Send(responseMessage); err != nil {
+		return fmt.Errorf("error sending response for execute request %s: %w", reqpb, err)
+	}
+
 	return nil
 }
 
-type callbackExecutableClient struct {
-	grpc capabilitiespb.CallbackExecutableClient
+type executableClient struct {
+	grpc capabilitiespb.ExecutableClient
 	*net.BrokerExt
 }
 
-func newCallbackExecutableClient(brokerExt *net.BrokerExt, conn *grpc.ClientConn) *callbackExecutableClient {
-	return &callbackExecutableClient{
-		grpc:      capabilitiespb.NewCallbackExecutableClient(conn),
+func newExecutableClient(brokerExt *net.BrokerExt, conn *grpc.ClientConn) *executableClient {
+	return &executableClient{
+		grpc:      capabilitiespb.NewExecutableClient(conn),
 		BrokerExt: brokerExt,
 	}
 }
 
-var _ capabilities.CallbackExecutable = (*callbackExecutableClient)(nil)
+var _ capabilities.Executable = (*executableClient)(nil)
 
-func (c *callbackExecutableClient) Execute(ctx context.Context, req capabilities.CapabilityRequest) (<-chan capabilities.CapabilityResponse, error) {
+func (c *executableClient) Execute(ctx context.Context, req capabilities.CapabilityRequest) (capabilities.CapabilityResponse, error) {
 	responseStream, err := c.grpc.Execute(ctx, pb.CapabilityRequestToProto(req))
 	if err != nil {
-		return nil, fmt.Errorf("error executing capability request: %w", err)
+		return capabilities.CapabilityResponse{}, fmt.Errorf("error executing capability request: %w", err)
 	}
 
-	resp, err := responseStream.Recv()
+	message, err := responseStream.Recv()
 	if err != nil {
-		return nil, fmt.Errorf("error waiting for ack: %w", err)
+		return capabilities.CapabilityResponse{}, fmt.Errorf("error waiting for ack: %w", err)
```

Comment:
We're not waiting for an ack anymore, I think?

Reply:
@ettec Can we also remove the Ack type from the ResponseMessage proto type -- there's actually probably some additional simplification we can do there.

Reply:
ah right, its waiting for the response, but its not an ack, I'll update the message

Reply:
removed the ResponseMessage and related code, simplifies things a bit as you say

Reply:
sorted the message